pkg/yttlibrary/assert.go

```diff
@@ -25,6 +26,20 @@ var (
 
 type assertModule struct{}
 
+func (b assertModule) Equals(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 2 {
+		return starlark.None, fmt.Errorf("expected two arguments")
+	}
+
+	expected, _ := core.NewStarlarkValue(args.Index(0)).AsString()
```

Comment:
going back at the motivating use case for this feature, [here](https://github.com/cloudfoundry/cf-for-k8s/blob/18b4f3f676a85e49e675120f5b720aab91ac3892/config/istio/gateway_test.star#L77-L81), the function allows types other than strings. We should expand this function to accept other types as well.

Reply:
I added a test for numbers. What other types could be passed in as arguments?

Reply:
It could get complex if we allowed all values such as maps, arrays, documents, etc. We do want to support comparing these types, however we would need to implement an algorithm to compare these nested structures which may be complex. So, to keep the scope of this story small I think we should only support scalar types for now. This is similar functionality to what currently exists with the function linked above. So that would be 
``` 
string, float64, int, int64, bool, nil
```
This can be accomplished by having a function that checks if the expected value is one of those types, and errors if not.

Reply:
> So, to keep the scope of this story small I think we should only support scalar types for now. 

unless we are planning to immediately follow up with a PR, i wouldnt mind seeing maps, arrays, yamlfragments etc in this PR.

Reply:
There was a conversation about this in [Slack ](https://kubernetes.slack.com/archives/CH8KCCKA5/p1638896293138700). We talked about encoding the value in yaml, which then could be compared and displayed in the error message.

Reply:
Added checks for maps, arrays and yamlfragments.